### PR TITLE
New Base32 wrapper and tests.

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description
Added wrapper for Base32 encoding, calls into Trezor implementation.  Updated 3 coins that use it.  Also added new tests.

## Types of changes
Minor refactoring.

### CODE REVIEW

File: src/Algorand/Address.cpp [hunk 1]
```diff
@@ -40,10 +37,10 @@ Address::Address(const std::string& string) {
     if (!isValid(string)) {
         throw std::invalid_argument("Invalid address string");
     }
-
-    const size_t dataSize = PublicKey::ed25519Size + checksumSize;
-    std::array<byte, dataSize> decoded;
-    base32_decode(string.data(), string.size(), decoded.data(), dataSize, BASE32_ALPHABET_RFC4648);
+    Data decoded;
+    if (!TW::Base32::decode(string, decoded)) {
```
Comment (hewigovens) [inline src/Algorand/Address.cpp]:
```suggestion
    if (!Base32::decode(string, decoded)) {
```

File: src/Algorand/Address.cpp [hunk 2]
```diff
@@ -18,14 +17,12 @@ bool Address::isValid(const std::string& string) {
     if (string.size() != encodedSize) {
         return false;
     }
-    const size_t dataSize = PublicKey::ed25519Size + checksumSize;
-    std::array<byte, dataSize> decoded;
-    // base32 decode
-    if (base32_decode(string.data(), string.size(), decoded.data(), dataSize, BASE32_ALPHABET_RFC4648) == nullptr) {
+    Data decoded;
+    if (!TW::Base32::decode(string, decoded)) {
```
Comment (hewigovens) [inline src/Algorand/Address.cpp]:
```suggestion
    if (!Base32::decode(string, decoded)) {
```

File: src/Base32.h [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    size_t outLen = base32_encoded_length(inLen) + 1;
+    char buf[outLen];
+    if (alphabet == nullptr) {
+        alphabet = BASE32_ALPHABET_RFC4648;
+    }
+    // perform the base32 encode
+    char* retval = base32_encode(val.data(), inLen, buf, outLen, alphabet);
+    assert(retval != nullptr);
+    // make sure there is a terminator ath the end
+    buf[outLen-1] = '\0';
```
Comment (hewigovens) [inline src/Base32.h]: clang-format will catch this
```suggestion
    buf[outLen - 1] = '\0';
```

**Comment by hewigovens** [comment]
you can add something like: Fixes / Closes #697 in pull request description, when the pr is merged, the corresponding issue is closed automatically

**Comment by laviol1980** [comment]
### __

**Pull request merged**